### ISSUE TITLE
wget link to download Train set and Test set? 

### ISSUE THREAD

**Issue by ghimiredhikura** [ox-vgg/vgg_face2#2]
Please help me to write wget link in order to download training and test set in here. 

[VGGFACE2](http://zeus.robots.ox.ac.uk/vgg_face2/)

![what_is_wget_link_train_test_data](https://user-images.githubusercontent.com/24696403/45596897-71fc6c00-b9fe-11e8-960d-50eb6b9f8039.png)

**Comment by WeidiXie** [comment]
Do you have any issues downloading the file with Chrome ?

BW
Weidi

**Comment by ghimiredhikura** [comment]
Yes I can download with chrome by mouse click. But I want to have automatic download script using wget or any other command. I tried this way but it is not working! 
```
wget 'zeus.robots.ox.ac.uk/vgg_face2/get_file?fname=vggface2_train.tar.gz'
```
Best, 
Deepak

**Comment by lishen-shirley** [comment]
Anonymous download is not allowed. You have to provide login and password. So it won't work just by using wget on the link. You can do it when creating a script that logs in and uses the cookies for the download.

**Comment by ghimiredhikura** [comment]
Yes I know that we have to provide login and pass. But with wget we can give login and pass in command line. That what I was looking for. Anyway, thanks for the comments.

**Comment by hi-ilkin** [comment]
If I want to upload dataset to my GCP server, I must download whole dataset on my pc and than upload it to server again?

**Comment by WeidiXie** [comment]
@huseynlilkin   If you want to download without browser, you can create a script.

**Comment by Huixxi** [comment]
Just  `wget http://balabala.tar.gz` directly won't work, because it is also required a lot request headers to be sent to the server to get the real dataset file. Try to "inspect" the "http://zeus.robots.ox.ac.uk/vgg_face2/" website page, click the "download link" to check the file "get_file?fname=vggface2_train.tar.gz"'s header, now, check the request header's content, then you can download the dataset with either:  
`wget --header='key1: value1'  --header='key2: value2'  ...   http://balabala.tar.gz`  
or write a python script with `import requests` but this method is too much memory cost.

**Comment by Sawatdatta** [comment]
Hello @huseynlilkin , is it possible for you to share the database if you have downloaded it at your end. please share via Dropbox or any other method. I am not able to completely download the database from original location, since the download stops due to network or power interruption and cannot be resumed. I have tried plenty of times.

If you have it downloaded please share the link.

Thank in advance.

**Comment by pricebenjamin** [comment]
Here's a Python 3 script for downloading the test set; it requires that you've already made an account on their website. I wanted to implement a progress bar, but the file size is not provided in the headers. Instead, the total number of bytes downloaded is displayed.

```python
import requests
import getpass
import sys

LOGIN_URL = "http://zeus.robots.ox.ac.uk/vgg_face2/login/"
FILE_URL = "http://zeus.robots.ox.ac.uk/vgg_face2/get_file?fname=vggface2_test.tar.gz"

print('Please enter your VGG Face 2 credentials:')
user_string = input('    User: ')
password_string = getpass.getpass(prompt='    Password: ')

payload = {
    'username': user_string,
    'password': password_string
}

session = requests.session()
r = session.get(LOGIN_URL)

if 'csrftoken' in session.cookies:
    csrftoken = session.cookies['csrftoken']
elif 'csrf' in session.cookies:
    csrftoken = session.cookies['csrf']
else:
    raise ValueError("Unable to locate CSRF token.")

payload['csrfmiddlewaretoken'] = csrftoken

r = session.post(LOGIN_URL, data=payload)

filename = FILE_URL.split('=')[-1]

with open(filename, "wb") as f:
    print(f"Downloading file: `{filename}`")
    r = session.get(FILE_URL, data=payload, stream=True)
    bytes_written = 0
    for data in r.iter_content(chunk_size=4096):
        f.write(data)
        bytes_written += len(data)
        MiB = bytes_written / (1024 * 1024)
        sys.stdout.write(f"\r{MiB:0.2f} MiB downloaded...")
        sys.stdout.flush()

print("\nDone.")
```

**Comment by Sawatdatta** [comment]
@pricebenjamin  Thanks for the script. Downloading the dataset.

**Comment by XingwXiong** [comment]
> Just `wget http://balabala.tar.gz` directly won't work, because it is also required a lot request headers to be sent to the server to get the real dataset file. Try to "inspect" the "http://zeus.robots.ox.ac.uk/vgg_face2/" website page, click the "download link" to check the file "get_file?fname=vggface2_train.tar.gz"'s header, now, check the request header's content, then you can download the dataset with either:
> `wget --header='key1: value1' --header='key2: value2' ... http://balabala.tar.gz`
> or write a python script with `import requests` but this method is too much memory cost.

Thanks for your help. @Huixxi 
I can download the dataset with :
```Shell
wget -c --header="cookie: cookie_string" \
     http://zeus.robots.ox.ac.uk/vgg_face2/get_file?fname=vggface2_train.tar.gz
```
We can find `cookie_string` in request header in web browser. Due to the large size of the file, it is recommended to download the file by resuming the breakpoint.

**Comment by dsp-avt** [comment]
@pricebenjamin is it okay to use/include your [python script](https://github.com/ox-vgg/vgg_face2/issues/2#issuecomment-461684945) in a project?

**Comment by pricebenjamin** [comment]
@dsp-avt, feel free! I hope it helps.

**Comment by scut-salmon** [comment]
The dataset downloaded by wget cannot unzip correctly, anyone meet the same problem? @XingwXiong @Huixxi 
![image](https://user-images.githubusercontent.com/20534039/71435445-fbcbd680-2723-11ea-8293-e91f646bfa5c.png)

**Comment by XingwXiong** [comment]
@scut-salmon  Sorry, I can't remember exactly. But I suggest you verify the MD5 checksum: `88813c6b15de58afc8fa75ea83361d7f`

**Comment by scut-salmon** [comment]
@XingwXiong Thanks, I add the `--no-cookies` to the original command and it can now unzip correctly.

**Comment by ysarathy99** [comment]
vgg_face2 is offline now :( 

https://github.com/ox-vgg/vgg_face2/issues/50

**Comment by ysarathy99** [comment]
workarounds:
- train with vgg face, download and use the utils (https://github.com/ndaidong/vgg-faces-utils)
- use pre-trained torch and caffe ones (https://www.robots.ox.ac.uk/~vgg/software/vgg_face/).. but a real pain if you want to 
 convert (say into TF).
